src/rfkill/rfkill.c

```diff
@@ -259,12 +269,31 @@ static int load_state(
         return 0;
 }
 
-static int save_state(
+static void save_state_queue_remove(
+                struct write_queue_item **write_queue,
+                int idx,
+                char *state_file) {
+
+        struct write_queue_item *item, *tmp;
+
+        LIST_FOREACH_SAFE(queue, item, tmp, *write_queue) {
+                if ((state_file && streq(item->file, state_file)) || idx == item->rfkill_idx) {
+                        log_debug("Canceled previous save state of '%s' to %s.", one_zero(item->state), item->file);
+                        LIST_REMOVE(queue, *write_queue, item);
+                        free(item->file);
+                        free(item);
```

Comment:
sorry I missed this on the first review, but I think we should probably have a destructor of some kind for this, i.e. write_queue_item_unlink() or so, that does the three steps above... after all, we have the very same three lines further down again, and we can share that then, and make it more readable